lib/WebService/Async/Segment.pm

```diff
@@ -0,0 +1,229 @@
+package WebService::Async::Segment;
+
+use strict;
+use warnings;
+
+use Net::Async::HTTP;
+use IO::Async::Loop;
+use Scalar::Util qw(blessed);
+use URI;
+use JSON::MaybeUTF8 qw(encode_json_utf8 decode_json_utf8);
+use Syntax::Keyword::Try;
+use Log::Any qw($log);
+use Date::Utility;
+
+use parent qw(IO::Async::Notifier);
+
+use WebService::Async::Segment::Customer;
+
+use constant SEGMENT_BASE_URL => 'https://api.segment.io/v1/';
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::Segment - Unofficial support for the Segment service
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+This class acts as a B<Future>-based async Perl wrapper for segment HTTP API.
+
+=cut
+
+=head1 METHODS
+
+=head2 new
+
+Class constuctor of Segment wrapper class with following argunments:
+
+parameters:
+
+=over 4
+
+=item * C<write_key> - the API token of a Segment source.
+
+=item * C<base_uri> - the base uri of the Segment host, primarily useful for setting up test mock servers.
+
+=back
+
+=cut
+
+sub _init {
```

Comment:
The documentation does not match the method name. The documentation should be for `->configure` instead.